src/WidgetBase.ts

```diff
@@ -210,28 +215,41 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 
 	public __setProperties__(originalProperties: this['properties']): void {
 		const properties = this._runBeforeProperties(originalProperties);
-		const changedPropertyKeys: string[] = [];
-		const allProperties = [ ...Object.keys(properties), ...Object.keys(this._properties) ];
-		const checkedProperties: string[] = [];
-		const diffPropertyResults: any = {};
 		const registeredDiffPropertyNames = this.getDecorator('registeredDiffProperty');
-		let runReactions = false;
+		const changedPropertyKeys: string[] = [];
+		const propertyNames = Object.keys(properties);
```

Comment:
just a reminder that on IE11 and any properties that are symbols, they will come back with `Object.keys`.  I don't know if you need symbol safe `Object.keys` here or not.

Reply:
We already used `Object.keys` within `__setProperties__` to get the property names, so this hasn't really changed (i just hoisted it, so I could share)